### PR TITLE
always set BUNDLE_WITHOUT

### LINKED ISSUE #2681: Bundler::GemfileNotFound thrown when bin/logstash isn't run from logstash root directory.
If you invoke logstash outside the root directory which it resides (either from source or when installed) you get the following result:

```
$ /tmp/logstash/bin/logstash -f /etc/logstash/conf.d --debug
Bundler::GemfileNotFound: Could not locate Gemfile or .bundle/ directory
                         root at /tmp/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.8.2/lib/bundler.rb:198
                  bundle_path at /tmp/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.8.2/lib/bundler.rb:101
  configure_gem_home_and_path at /tmp/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.8.2/lib/bundler.rb:420
                    configure at /tmp/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.8.2/lib/bundler.rb:92
                   definition at /tmp/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.8.2/lib/bundler.rb:153
                        setup at /tmp/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.8.2/lib/bundler.rb:118
               bundler_setup! at /tmp/logstash/lib/logstash/environment.rb:106
                       (root) at /tmp/logstash/lib/logstash/runner.rb:8
```

If I `cd` into `/tmp/logstash` (where I have master checked out) and then run the command it works fine.

The problem is that https://github.com/elasticsearch/logstash/blob/9561ade0353c4f12b3ba39edfd43a1d75e1a7748/lib/logstash/environment.rb#L91 detects that `[logstash_root]/.bundle/config` exists, however there isn't ever an entry for `BUNDLE_GEMFILE` so unless you are in `logstash_root` Bundler cannot find `Gemfile`.  This can be fixed by moving `ENV["BUNDLE_GEMFILE"] = LogStash::Environment::GEMFILE_PATH` outside the `unless`.

It's hard for me to tell, being unfamiliar with Bundler as much, where BUNDLE_GEMFILE would be written to the config file.

### PR DESCRIPTION
**Pull request by colinsurprenant**
fixes #2681 
relates to #2674

### CODE REVIEW

**Comment by ph** [comment]
Lgtm,  I did a similar test on Windows.

**Comment by colinsurprenant** [comment]
@ph Lgtm? I expect nothing less than a LGTM!

**Comment by ph** [comment]
This was an android lgtm

File: lib/logstash/environment.rb [hunk 1]
```diff
@@ -90,9 +90,11 @@ def bundler_setup!
       # if there's no .bundle/config setup the runtime environment.
       unless ::File.exists?(BUNDLE_CONFIG_PATH)
         ENV["BUNDLE_PATH"] = LogStash::Environment::BUNDLE_DIR
-        ENV["BUNDLE_GEMFILE"] = LogStash::Environment::GEMFILE_PATH
         ENV["BUNDLE_WITHOUT"] = "development"
       end
+      # force BUNDLE_GEMFILE since Bundler does not store it in its ./bundle/config. this is required otherwise Bundler will look for the Gemfile in the CWD
+      # and will crash when invoking logstash outside its home dir.
+      ENV["BUNDLE_GEMFILE"] = LogStash::Environment::GEMFILE_PATH
```
Comment (jordansissel) [inline lib/logstash/environment.rb]: BUNDLERRRRRRRR

**Comment by jordansissel** [comment]
lgtm

File: lib/logstash/environment.rb (hunk 1 above)
Comment (ph) [inline lib/logstash/environment.rb]: First rule of bundler is to never talk about bundler

**Comment by colinsurprenant** [comment]
:shipit:

**Pull request closed without merging**